### PR TITLE
Feature/molecule photo uploader content

### PR DESCRIPTION
**Pull request by AitorRodriguez990**
#### `❓ Ask`

**TASK**:  [#2049](https://github.com/SUI-Components/sui-components/discussions/2049)

### Types of changes

- [ ] 🪲 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [x] 📷 Demo
- [ ] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles

### Description, Motivation and Context

On the Real Estate vertical we need to add labels on photos, so I iterate the `MoleculePhotoUploader` component to be able to insert content inside.

### Screenshots - Animations

image

### CODE REVIEW

**Review by midudev — APPROVED** [review]

File: components/molecule/photoUploader/src/styles/settings.scss [hunk 1]
```diff
@@ -43,6 +43,7 @@ $jc-photo-uploader: center !default;
 $p-photo-uploader-action-buttons: $p-m 0 !default;
 $p-photo-uploader-desktop: $p-xl !default;
 $p-photo-uploader: $p-m !default;
+$p-photo-uploader-content: $p-m !default;
```
Comment (andresin87) [inline components/molecule/photoUploader/src/styles/settings.scss]: not necessary, we must offer an empty canvas and delegate the spacing margins-paddings L&F to the consumer of the space

File: components/molecule/photoUploader/src/styles/index.scss [hunk 2]
```diff
@@ -206,6 +205,10 @@ $skeleton-class: '#{$base-class}-skeleton';
       }
     }
 
+    &-content {
+      padding: $p-photo-uploader-content;
```
Comment (andresin87) [inline components/molecule/photoUploader/src/styles/index.scss]: not necessary

File: components/molecule/photoUploader/src/ThumbCard/index.js [hunk 3]
```diff
@@ -52,6 +54,7 @@ const ThumbCard = ({
           <img src={image.preview} className={imageThumbClass} />
         )}
       </div>
+      <div className={CONTENT_THUMB_CARD_CLASS_NAME}>{content()}</div>
```
Comment (andresin87) [inline components/molecule/photoUploader/src/ThumbCard/index.js]:
```suggestion
      {content !== undefined && <div className={CONTENT_THUMB_CARD_CLASS_NAME}>{content()}</div>}
```

File: components/molecule/photoUploader/src/PhotosPreview/index.js [hunk 4]
```diff
@@ -173,6 +174,7 @@ const PhotosPreview = ({
             callbackDeleteItem={_deleteItem}
             callbackRetryUpload={_retryUpload}
             callbackRotateItem={_rotateItem}
+            content={() => content(file, index)}
```
Comment (andresin87) [inline components/molecule/photoUploader/src/PhotosPreview/index.js]:
```suggestion
            content={() => content({file, index})}
```

File: components/molecule/photoUploader/demo/articles/InitialPhotosArticle.js [hunk 5]
```diff
@@ -24,19 +28,39 @@ import {
...
   _deleteIcon,
   _dragPhotosIcon,
   _infoIcon,
+  _labelsArrowIcon,
   _rejectPhotosIcon,
   _retryErrorPhotosIcon,
   _rotateIcon
 } from '../icons.js'
 
+const _content = (image, index) => {
```
Comment (andresin87) [inline components/molecule/photoUploader/demo/articles/InitialPhotosArticle.js]:
```suggestion
const _content = ({file: image, index}) => {
```

File: components/molecule/photoUploader/src/PhotosPreview/index.js (hunk 4 above)
Comment (andresin87) [reply components/molecule/photoUploader/src/PhotosPreview/index.js]: creo que tb funcionaria
`content={content.bind(undefined, {file, index})}`

**Review by andresin87 — CHANGES_REQUESTED** [review]
👏👏👏

File: components/molecule/photoUploader/src/styles/index.scss (hunk 2 above)
Comment (andresin87) [reply components/molecule/photoUploader/src/styles/index.scss]: I think this might be in demos styles and verticals

**Comment by andresin87** [comment]
The deployed demo is not storing the changes through the image selectors... Is it right?

**Comment by andresin87** [comment]
Better add a new demo for documenting the content prop copy-pasting the initialProtos one, not reusing it and also document the prop usage in that new article

**Comment by AitorRodriguez990** [comment]
Thank you @andresin87 👌 

I applied your feedback, can you review it, please? 

Now, I only have a problem: I can't change the array values, like the `label`, because the component is a controlled component and has its own state. Can you help me here, please?

File: components/molecule/photoUploader/src/index.js [hunk 6]
```diff
@@ -387,6 +389,9 @@ MoleculePhotoUploader.propTypes = {
    */
   callbackUploadPhoto: PropTypes.func,
 
+  /** Component to render inside content space */
+  content: PropTypes.func,
```
Comment (andresin87) [inline components/molecule/photoUploader/src/index.js]: I've been thinking about that. Maybe content is too much ambiguous. Think about `fileChildren` or `childenFile` or something similar

**Comment by andresin87** [comment]
With Content - Select example might work keeping the user selected value

**Review by andresin87 — APPROVED** [review]

**Pull request merged**